components/ws-manager/pkg/manager/status.go

```diff
@@ -509,9 +502,15 @@ func (m *Manager) extractStatusFromPod(result *api.WorkspaceStatus, wso workspac
 			}
 		}
 
-		if wso.IsWorkspaceHeadless() {
-			// headless workspaces don't expose a public service and thus cannot be asked about their status.
-			// once kubernetes reports the workspace running, so do we.
+		tpe, err := wso.WorkspaceType()
+		if err != nil {
+			log.WithError(err).Warn("cannot determine workspace type - assuming this is a regular")
+			tpe = api.WorkspaceType_REGULAR
+		}
+
+		if tpe == api.WorkspaceType_GHOST {
```

Comment:
At this point the term headless has become a bit of a misnomer, because they do serve a service.
I reckon we should make the prebuild case explicit here rather than expect all but Ghost workspaces to offer a service.

Reply:
I'm fine with the code either way, but IMO ghosts are the only real exception left because they use a completely different code to run (in supervisor) compared other workspaces types.

"headless workspaces" still fits for me, because we do not expect them to have a frontend connected. I also looked into decoupling the "StatusService" in supervisor from the rest of the supervisor machinery, but it turned out to be more code (which would require further discussion).